timesketch/ui/static/components/explore/explore-filter-directive.js

```diff
@@ -68,8 +70,37 @@ limitations under the License.
                     scope.meta.noisy = false;
                 }
 
-            }
-        }
+                scope.parseFilterDate = function(datevalue){  
+                    if (datevalue != null) {
+                        var datetimetemplate="YYYY-MM-DDTHH:mm:ss";
+                        //Parse out 'T' date time seperator needed by ELK but not by moment.js
+                        datevalue=datevalue.replace(/T/g,' ');
+                        //Parse offset given by user. Eg. +-10m
+                        var offsetRegexp = /(.*?)(-|\+|\+-|-\+)(\d+)(y|d|h|m|s|M|Q|w|ms)/g;
+                        var match = offsetRegexp.exec(datevalue);
+                        if (match != null) {
+                            match[1] = moment(match[1],"YYYY-MM-DD HH:mm:ssZZ");
```

Comment:
It's a bit hard to follow here without looking at the regexp. Could you put the mathed expressions in variables perhaps? Just to make it easier to read.